app/views/search/search.html.erb

```diff
@@ -3,31 +3,31 @@
 <%= form_tag (params[:option] == 'graphs' ? search_path(anchor: "graphs") : search_path), method: "get" do |f| %>
   <div class="form-group">
     <%= label_tag :title %>
-    <% if user_signed_in? %>
+    <% if user_signed_in? and current_user.can_use_regex_search? %>
       <%= check_box_tag :title_is_regex, 1, params[:title_is_regex] %> <span class="text-muted">regex</span>
       <%= check_box_tag :title_is_inverse_regex, 1, params[:title_is_inverse_regex] %> <span class="text-muted">invert</span>
     <% end %>
     <%= text_field_tag :title, params[:title], placeholder: "*", class: "form-control" %>
   </div>
   <div class="form-group">
     <%= label_tag :body %>
-    <% if user_signed_in? %>
+    <% if user_signed_in? and current_user.can_use_regex_search? %>
       <%= check_box_tag :body_is_regex, 1, params[:body_is_regex] %> <span class="text-muted">regex</span>
       <%= check_box_tag :body_is_inverse_regex, 1, params[:body_is_inverse_regex] %> <span class="text-muted">invert</span>
     <% end %>
     <%= text_field_tag :body, params[:body], placeholder: "*", class: "form-control" %>
   </div>
   <div class="form-group">
     <%= label_tag :username %>
-    <% if user_signed_in? %>
+    <% if user_signed_in? and current_user.can_use_regex_search? %>
       <%= check_box_tag :username_is_regex, 1, params[:username_is_regex] %> <span class="text-muted">regex</span>
       <%= check_box_tag :username_is_inverse_regex, 1, params[:username_is_inverse_regex] %> <span class="text-muted">invert</span>
     <% end %>
     <%= text_field_tag :username, params[:username], placeholder: "*", class: "form-control" %>
   </div>
   <div class="form-group">
     <%= label_tag :why %>
-    <% if user_signed_in? %>
```

Comment:
These can be simplified to `if current_user&.can_user_regex_search?`

Reply:
Whoa, fancy!